packages/eas-cli/src/project/projectUtils.ts

```diff
@@ -36,21 +36,11 @@ export function getProjectAccountName(exp: ExpoConfig, user: Actor): string {
   }
 }
 
-/**
- * @deprecated - prefer using the account that definitively owns the project by
- *               fetching it via the App.ownerAccount GraphQL field.
- */
-export function getUsername(exp: ExpoConfig, user: Actor): string | undefined {
+export function getUsernameForActor(user: Actor): string | undefined {
   switch (user.__typename) {
     case 'User':
       return user.username;
     case 'Robot':
-      // owner field is necessary to run `expo prebuild`
-      if (!exp.owner) {
```

Comment:
A bit of context on why it was this way:

When you are running prebuild, expo-cli is using the owner field for some stuff, and if it is not present it fallbacks to the username. In the case of EAS Build, we are running prebuild with unauthenticated expo-cli on our own servers, so we are passing this username as a fallback, in case the owner is not present.
Even If everything is migrated already and nothing in prebuilds process or in SDK depends on the owner field, we still need to support older projects.

With the current state of this PR if someone is building using a bot account, does not have an owner field in app.json, and has a project old enough to relay on the owner field they will get the same result as if they ran prebuild without logging in.

The approach I would recommend is to:
- for older projects that still use the owner field(no idea if there is a way to detect that), if you don't want that check here, then add validation somewhere in the build process e.g https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/build/runBuildAndSubmit.ts#L260
- for new projects, I'm assuming that we don't need that validation, but if it's hard to detect I don't see anything wrong with throwing that error for robot users and requiring an explicit owner field.